### PR TITLE
Dogs now flee from mobs with tongs

### PR DESCRIPTION
**Pull request by optimumtact**
AI dogs with the dog controller behaviour will flee from a target with tongs in hand.

Untested because I literally cannot play byond.

## Why It's Good For The Game
https://www.youtube.com/watch?v=cXIAZtwvgz0

## Changelog
:cl: oranges
add: Dogs now react to centrist grillers more realistically
/:cl:

### CODE REVIEW

File: code/datums/ai/basic_mobs/targetting_datums/with_object.dm [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+		// Leaving object type as null will make this basically a noop
+		return
+	src.object_type_path = object_type_path
+
+///Returns true or false depending on if the target can be attacked by the mob
+/datum/targetting_datum/basic/holding_object/can_attack(mob/living/living_mob, atom/target, vision_range)
+	if (object_type_path == null)
+		return FALSE // no op
+	// Check if our parent behaviour agrees we can attack this target, ignoring factions
+	var/can_attack = ..(living_mob, target, vision_range, FALSE)
```
Comment (Timberpoes) [inline code/datums/ai/basic_mobs/targetting_datums/with_object.dm:30]: The `var/can_attack = ..(living_mob, target, vision_range, FALSE)` could potentially cause issues if the parameters ever change. 

Are you willing to change the proc params to `/datum/targetting_datum/basic/holding_object/can_attack(mob/living/living_mob, atom/target, vision_range, check_faction = FALSE)` and then just call parent without args?

Not a blocking suggestion, just a stylistic one.

**Review by Timberpoes — CHANGES_REQUESTED** [review]

File: code/datums/ai/basic_mobs/targetting_datums/with_object.dm (hunk 1 above)
Comment (optimumtact) [reply code/datums/ai/basic_mobs/targetting_datums/with_object.dm:30]: it's cringe and gross but okay.

Comment (Timberpoes) [reply code/datums/ai/basic_mobs/targetting_datums/with_object.dm:30]: o7

**Review by Jacquerel — CHANGES_REQUESTED** [review]
I checked it out and it doesnt work but it _almost_ does
and one of them is just because we put a footgun in there

**Comment by Kapu1178** [comment]
SPEED MERGE

**Pull request merged**